### PR TITLE
refactor: split voter_olas.py into three focused services

### PR DESCRIPTION
**Pull request by mkrueger12**
## PR Type

feature

## Summary

This PR implements a test-driven refactor to split the monolithic `voter_olas.py` file into three focused services following the single responsibility principle:

| File Name | Change Summary |
| --- | --- |
| backend/config.py | Added OLAS configuration settings for new services (Safe addresses, RPC endpoints, store path) |
| backend/main.py | Added imports and initialization for three new services in lifespan context |
| backend/services/safe_service.py | **New file**: SafeService class handling all Safe multi-signature wallet operations |
| backend/services/activity_service.py | **New file**: ActivityService class tracking OLAS compliance and daily activity requirements |
| backend/services/voting_service.py | **New file**: VotingService class managing Snapshot DAO voting operations |
| backend/tests/test_safe_service.py | **New file**: Comprehensive unit tests for SafeService (12 tests) |
| backend/tests/test_activity_service.py | **New file**: Comprehensive unit tests for ActivityService (17 tests) |
| backend/tests/test_voting_service.py | **New file**: Comprehensive unit tests for VotingService (15 tests) |
| backend/tests/test_service_integration.py | **New file**: Integration tests verifying services work together (5 tests) |

## Architecture Benefits

- **Single Responsibility**: Each service has one clear purpose
- **Separation of Concerns**: Safe transactions, activity tracking, and voting are independent
- **Testability**: 49 total tests with 100% pass rate, comprehensive mocking
- **Maintainability**: Clear boundaries make code easier to understand and modify
- **Scalability**: Services can be enhanced independently

## Test Coverage

- **49 total tests** across all three services and integration tests
- **Test-driven development** approach where tests were written first
- **100% test pass rate** with comprehensive unit and integration testing
- **Integration tests** verify services work together correctly

## Breaking Changes

No breaking changes - this is a pure refactor that maintains all existing functionality while improving architecture.

## Edge Cases for Reviewer

No edge cases found - this is a pure refactor with comprehensive test coverage ensuring all functionality is preserved.

## Reviewer Test Plan

- [ ] Run all new tests: `uv run pytest tests/test_safe_service.py tests/test_activity_service.py tests/test_voting_service.py tests/test_service_integration.py`
- [ ] Verify no regressions in existing tests: `uv run pytest tests/test_tally_service.py tests/test_ai_service.py`
- [ ] Check services can be imported without errors: `uv run python -c "from services.safe_service import SafeService; from services.activity_service import ActivityService; from services.voting_service import VotingService"`
- [ ] Verify application startup works with new services
- [ ] Confirm all three services share the same EOA account from ethereum_private_key.txt
- [ ] Test ActivityService can coordinate with SafeService for compliance transactions

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: backend/services/safe_service.py [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+        """
+        safe_address = self.safe_addresses.get(chain)
+        if not safe_address:
+            raise ValueError(f"No Safe address configured for chain: {chain}")
+        
+        safe_address = Web3.to_checksum_address(safe_address)
+        eth_client = EthereumClient(self.rpc_endpoints[chain])
+        safe_instance = Safe(safe_address, eth_client)
+        
+        safe_tx = safe_instance.build_multisig_tx(
```
Comment (mkrueger12) [inline backend/services/safe_service.py:264]: @claude why do the values of this method reference themselves?
